src/bin/inx-chronicle/main.rs

```diff
@@ -44,6 +44,27 @@ async fn main() -> eyre::Result<()> {
         ByteSize::b(db.size().await?)
     );
 
+    match db
+        .collection::<ApplicationStateCollection>()
+        .get_last_migration()
+        .await?
+        .as_deref()
+    {
+        None => {
+            info!("Setting migration version to {}", migrations::LATEST_VERSION);
+            db.collection::<ApplicationStateCollection>()
+                .set_last_migration(migrations::LATEST_VERSION)
+                .await?;
+        }
+        Some(migrations::LATEST_VERSION) => (),
+        Some(v) => {
+            eyre::bail!(
+                "Invalid database migration version {}, please run the `migrate` command.",
+                v
+            );
+        }
+    }
```

Comment:
Can we refactor this into a method (`handle_migration`) and inline it? Reason being that it takes up quite a few lines in our main function, but it shouldn't since it runs only in rare cases.

Reply:
I would love for stuff to be as self-contained as possible, so I would like that. But we can open an issue for it if you don't want to do that now.

Reply:
You're right that is subjective like many things in the end, but for those "controller" like functions (call A, then B, then C, then D, ...) it's nice to refactor the logic out into small, self-describing sub-routines that can (and probably will automatically) be inlined. By picking good names one gets a quick idea what `main` is actually doing without the reader having to parse undocumented code.